OpenRA.Mods.Common/UpdateRules/Rules/20210321/AddControlGroups.cs

```diff
@@ -0,0 +1,29 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddControlGroups : UpdateRule
+	{
+		public override string Name => "Add new ControlGroups trait.";
+
+		public override string Description => "A new trait ControlGroups was added, splitting logic away from Selection.";
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			if (actorNode.ChildrenMatching("Selection").Any())
```

Comment:
My bad. I didn't see that the logic was moved from Selection to ControlGroups trait.

In that case 
```suggestion
			if (actorNode.ChildrenMatching("Selection").Any(t => !t.IsRemoval()))
```

Reply:
I don't think we allow more than one `Selection` to exist on an actor. So really I don't really see a problem in just adding `ControlGroups` trait automatically

Reply:
Since `Selection` has no values I think we can assume that it is never overwritten. So we could just add the `ControlGroups` and it should not cause issues 99% of the time